### ISSUE TITLE
ERL-1367: Add maps:merge/3 to allow for a conflict resolution function

### ISSUE THREAD

**Issue by OTP-Maintainer** [erlang/otp#4415]
Original reporter: `JIRAUSER16800`
Affected version: `Not Specified`
Component: `stdlib`
Migrated from: https://bugs.erlang.org/browse/ERL-1367

---

```
When merging two maps, `M1` and `M2`, it is sometimes useful to be able to run a function to handle a case where there is a key present in both maps, rather than taking the value from `M2` unilaterally. For example to sum the values for a given key:

 

{{M1 = #\{"total" => 2},}}

{{M2 = #\{"total" => 2}, }}

{{Fun = fun(Key, Value1, Value2) -> Value1 + Value2 end,}}

{{maps:merge(M1, M2, Fun).}}

% #\{"total" := 4}

 
```

**Comment by josevalim** [comment]
This can be closed as it has been merged and will be available in OTP 24. :)

**Comment by garazdawi** [comment]
merged as #2797
